From b23b7dab1d540b0710fcb9ded1c6256a49844906 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B8rn=20Forsman?= <bjorn.forsman@gmail.com>
Date: Wed, 20 Aug 2014 22:22:00 +0200
Subject: [PATCH] Fix wrong "isProcess" logic

Stopmotion wrongly thinks that uvccapture should be run as a daemon,
even though configuration for uvccapture has no "daemon-like" command
line to be run (according to "preferences"). The result is an error
popup instead of video/image grabbing.

This brings back the "isProcess" logic that was in stopmotion v0.7.2,
because it seems to work, while the current logic (v0.8.0) seems to
fail.
---
 src/presentation/frontends/qtfrontend/frameview.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/presentation/frontends/qtfrontend/frameview.cpp b/src/presentation/frontends/qtfrontend/frameview.cpp
index e44dca7..d2c41fd 100644
--- a/src/presentation/frontends/qtfrontend/frameview.cpp
+++ b/src/presentation/frontends/qtfrontend/frameview.cpp
@@ -270,7 +270,7 @@ bool FrameView::on() {
 	Preference device(QString("device%1")
 			.arg(activeDev).toLatin1().constData(), "");
 	QString pre = QString(prepoll.get()).replace("$VIDEODEVICE", device.get());
-	bool isProcess = startDaemon.get();
+	bool isProcess = (strcmp(startDaemon.get(), "") == 0) ? false : true;
 
 	bool isCameraReady = true;
 	this->grabber = new CommandLineGrabber(capturedFile.path(), isProcess);
-- 
2.0.2

